leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -159,18 +164,60 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args) th
 
     private RuntimeException handleDelegateThrewException(
             LeadershipToken leadershipToken, InvocationTargetException exception) throws Exception {
-        if (exception.getCause() instanceof ServiceNotAvailableException
-                || exception.getCause() instanceof NotCurrentLeaderException) {
-            leadershipStateManager.handleNotLeading(leadershipToken, exception.getCause());
+        Throwable cause = exception.getCause();
+        if (cause instanceof SuspectedNotCurrentLeaderException) {
+            handleSuspectedNotCurrentLeader(leadershipToken, (SuspectedNotCurrentLeaderException) cause);
+        }
+        if (cause instanceof ServiceNotAvailableException || cause instanceof NotCurrentLeaderException) {
```

Comment:
`Condition 'cause instanceof NotCurrentLeaderException' is always 'false' when reached` as `NotCurrentLeaderException extends ServiceNotAvailableException` 

```suggestion
        if (cause instanceof ServiceNotAvailableException) {
```

Reply:
(Yep I didn't touch this at the time as I wasn't really modifying that, but that's right.)